### DISCUSSION TITLE
Change Proposal: CLI syntax

### DISCUSSION THREAD

**Discussion by rical** [kernelkit/infix#9] [Ideas]
Here's some suggestions of (breaking) syntax changes based on the first impression of a new user. I don't know the background or reasoning for the current syntax. So take it for what it is, suggestions. The current syntax is completely fine as it is.

### Configure context
`abort - Abandon candidate` ->
`abort - Discard config candidate and leave current context`

`check    - Validate candidate` ->
`validate - Validate config candidate`

`commit - Commit current candidate to running-config` ->
`commit - Commit config candidate to running-config`

`diff - Summarize uncommitted changes`
`diff - Show changes in config candidate`

`exit - Ascend to the parent configuration node, or abort (from top)` ->
Remove, at least from sub contexts such as configure (abort makes much more sense).

`leave - Finalize candidate and apply to running-config` ->
`leave - Commit config candidate and leave current context`

`rollback  - Restore candidate to running-config` ->
`reset     - Reset config candidate to running-config`

`top - Ascend to the configuration root` ->
Remove, confusing, what happens to the config?

`up - Ascend to the parent configuration node` ->
Remove, confusing, what happens to the config?

`no - Alias for delete` ->
Remove, delete or unset makes more sense

`delete - Delete configuration setting(s)` ->
`unset - Revert setting to default value`

**Comment by troglobit** [comment]
Interesting, @wkz and I had an intensive discussion about this last week.  Before I respond to specific suggestions, here's some background that might be useful.

The CLI we've based ours on (klish) is a reimplementation of the JunOS CLI by Juniper Networks.  It comes with many odd little things that we've tried to both keep but also put our own flavor on, which is inspired by both the Westermo CLI, Cisco and HP Procurve.

Since the CLI behaves a lot like JunOS we decided to keep most of its features as-is, the syntax can be found here:
 - https://www.juniper.net/documentation/us/en/software/jatp/jatp-cli-reference/topics/topic-map/jatp-all-in-one-cli-commands.html
 - https://www.juniper.net/documentation/us/en/software/junos/cli/cli.pdf

Have a look and let us know what you think.  In any case, there's definitely room for improvement, in particular to the online documentation of each command.
